Content.Server/HotPotato/HotPotatoSystem.cs

```diff
@@ -0,0 +1,117 @@
+using Content.Shared.MobState.Components;
+using Content.Shared.Popups;
+using Robust.Shared.Containers;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+
+namespace Content.Server.HotPotato;
+
+/// <summary>
+/// This handles...
+/// </summary>
+public sealed class HotPotatoSystem : EntitySystem
+{
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<HotPotatoComponent, AfterInteractEvent>(OnInteractUsing);
+        SubscribeLocalEvent<HotPotatoComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<HotPotatoComponent, ContainerGettingRemovedAttemptEvent>(OnRemoveAttempt);
+        SubscribeLocalEvent<HotPotatoComponent, TriggerEvent>(OnTrigger);
+    }
+
+    private void OnTrigger(EntityUid uid, HotPotatoComponent component, TriggerEvent args)
+    {
+        if (component.IsDud)
+        {
+            RemComp<UnremoveableComponent>(component.Owner);
+            var newItem = Spawn(component.TurnInto, Transform((component.Owner)).MapPosition);
+
+            _audioSystem.PlayPvs(component.DudSound, newItem);
+            _popupSystem.PopupEntity(Loc.GetString("hot-potato-dud", ("item", component.Owner), ("newitem", newItem)), newItem, Filter.Pvs(newItem), PopupType.Medium);
+
+            QueueDel(component.Owner);
+        }
+    }
+
+    private void OnRemoveAttempt(EntityUid uid, HotPotatoComponent component, ContainerGettingRemovedAttemptEvent args)
+    {
+        if (HasComp<UnremoveableComponent>(component.Owner))
+            _popupSystem.PopupEntity(Loc.GetString("hot-potato-drop-fail", ("item", component.Owner)), uid, Filter.Entities(args.Container.Owner), PopupType.MediumCaution);
+    }
+
+    private void OnUseInHand(EntityUid uid, HotPotatoComponent component, UseInHandEvent args)
+    {
+        if (!component.IsActivated)
+        {
+            EnsureComp<UnremoveableComponent>(component.Owner);
+            _popupSystem.PopupEntity(Loc.GetString("hot-potato-activate", ("user", args.User), ("item", component.Owner)), args.User, Filter.Pvs(args.User), PopupType.LargeCaution);
+            component.IsActivated = true;
+        }
+    }
+
+    private void OnInteractUsing(EntityUid uid, HotPotatoComponent component, AfterInteractEvent args)
+    {
+        if (!component.IsActivated)
+            return;
+
+        if (args.User == args.Target)
+            return;
+
+        var currentTime = _gameTiming.CurTime;
+        if (currentTime < component.CooldownEnd)
+            return;
+
+        component.LastUseTime = currentTime;
+        component.CooldownEnd = component.LastUseTime + TimeSpan.FromSeconds(component.CooldownTime);
```

Comment:
instead of tracking this yourself just use `ItemCooldown` and `UseDelay` which already exist, and check the cooldown on that comp instead